File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(a, b) {
    res=[];
    let i=0,j=0;
    while(i<a.length && j<b.length){
        if(a[i]<b[j]){
            res.push(a[i++]);
        }
        else{
            res.push(b[j++]);
        }
    }
    while(i<a.length){
        res.push(a[i++]);
    }
    while(j<b.length){
        res.push(b[j++]);
    }
    let n=res.length;
    if(n%2==0){
        return (res[n/2]+res[n/2-1])/2;
    }
    return res[Math.floor(n/2)];
};

File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int left = 0, right = 0, ind = 0;
        int temp[] = new int[nums1.length+nums2.length];
        while(left<nums1.length && right < nums2.length){
            if(nums1[left]>=nums2[right]) temp[ind++] = nums2[right++];
            else temp[ind++] = nums1[left++];
        }
        while(left<nums1.length) temp[ind++] = nums1[left++];
        while(right<nums2.length) temp[ind++] = nums2[right++];

        int n = temp.length;
        if(n%2!=0) return temp[n/2];
        else return (double)(temp[n/2 - 1]+temp[n/2])/2;
    }
}